From a47a2d28c75a87f72295c139518aa75506093617 Mon Sep 17 00:00:00 2001
Message-Id: <patch-v4-02.34-a47a2d28c75-20211117T101025Z-avarab@gmail.com>
In-Reply-To: <cover-v4-00.34-00000000000-20211117T101025Z-avarab@gmail.com>
References: <cover-v3-00.23-00000000000-20211116T114334Z-avarab@gmail.com>
	<cover-v4-00.34-00000000000-20211117T101025Z-avarab@gmail.com>
From: =?UTF-8?q?=C3=86var=20Arnfj=C3=B6r=C3=B0=20Bjarmason?=
 <avarab@gmail.com>
Date: Thu, 25 Mar 2021 11:22:23 +0100
Subject: [PATCH v4 02/34] add docs: include a CONFIGURATION section
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Add a CONFIGURATION section to the documentation for git-add. Since
we'll want to use the same blurb as I added in b6a8d09f6d8 (gc docs:
include the "gc.*" section from "config" in "gc", 2019-04-07) let's
make it a template in preparation for repeating this pattern in other
command documentation.

Signed-off-by: Ævar Arnfjörð Bjarmason <avarab@gmail.com>
---
 Documentation/Makefile                            | 1 +
 Documentation/git-add.txt                         | 7 +++++++
 Documentation/git-gc.txt                          | 4 +---
 Documentation/includes/cmd-config-section-all.txt | 3 +++
 4 files changed, 12 insertions(+), 3 deletions(-)
 create mode 100644 Documentation/includes/cmd-config-section-all.txt

diff --git a/Documentation/Makefile b/Documentation/Makefile
index d16b653394c..67f1866eee6 100644
--- a/Documentation/Makefile
+++ b/Documentation/Makefile
@@ -50,6 +50,7 @@ MAN7_TXT += gitworkflows.txt
 HOWTO_TXT += $(wildcard howto/*.txt)
 
 DOC_DEP_TXT += $(wildcard *.txt)
+DOC_DEP_TXT += $(wildcard includes/*.txt)
 DOC_DEP_TXT += $(wildcard config/*.txt)
 
 ifdef MAN_FILTER
diff --git a/Documentation/git-add.txt b/Documentation/git-add.txt
index 11eb70f16c7..36da8bf9b51 100644
--- a/Documentation/git-add.txt
+++ b/Documentation/git-add.txt
@@ -431,6 +431,13 @@ they will make the patch impossible to apply:
 * deleting context or removal lines
 * modifying the contents of context or removal lines
 
+CONFIGURATION
+-------------
+
+include::includes/cmd-config-section-all.txt[]
+
+include::config/add.txt[]
+
 SEE ALSO
 --------
 linkgit:git-status[1]
diff --git a/Documentation/git-gc.txt b/Documentation/git-gc.txt
index afb2fef1d04..2d2b142058e 100644
--- a/Documentation/git-gc.txt
+++ b/Documentation/git-gc.txt
@@ -145,9 +145,7 @@ linkgit:githooks[5] for more information.
 CONFIGURATION
 -------------
 
-Everything below this line in this section is selectively included
-from the linkgit:git-config[1] documentation. The content is the same
-as what's found there:
+include::includes/cmd-config-section-all.txt[]
 
 include::config/gc.txt[]
 
diff --git a/Documentation/includes/cmd-config-section-all.txt b/Documentation/includes/cmd-config-section-all.txt
new file mode 100644
index 00000000000..296a239f2ab
--- /dev/null
+++ b/Documentation/includes/cmd-config-section-all.txt
@@ -0,0 +1,3 @@
+Everything below this line in this section is selectively included
+from the linkgit:git-config[1] documentation. The content is the same
+as what's found there:
-- 
2.34.0.796.g126c2ced91c

